File: Testing/Code for Testing/Early Basic Syntax Testing/bounded_array/test_bounded_array.py
from dataclasses import dataclass, field

def _ul_sort(arr, n): arr[:n] = sorted(arr[:n])

class _ul_Exception(Exception):
    def __init__(self, type_name, message):
        self.type_name = type_name
        self.message = message
        super().__init__(message)


def main():
    nums = [10, 20, 30, 40, 50]
    print(nums[0])
    print(nums[4])
    buf = []
    buf[0] = 100
    buf[1] = 200
    buf[2] = 300
    print(buf[0])
    print(buf[2])
    vals = [1, 2, 3]
    print(vals[0])
    print(vals[2])
    return 0


if __name__ == "__main__":
    main()

File: Testing/Code for Testing/Early Basic Syntax Testing/bounded_array/test_bounded_array.js

function main() {
    let nums = [10, 20, 30, 40, 50];
    console.log(nums[0]);
    console.log(nums[4]);
    let buf = new Array(3).fill(0);
    buf[0] = 100;
    buf[1] = 200;
    buf[2] = 300;
    console.log(buf[0]);
    console.log(buf[2]);
    let vals = [1, 2, 3];
    console.log(vals[0]);
    console.log(vals[2]);
    return 0;
}

function __ul_sort(arr, n) { const sub = arr.slice(0, n).sort((a, b) => a < b ? -1 : a > b ? 1 : 0); for (let i = 0; i < n; i++) arr[i] = sub[i]; }

main();
